src/compiler/types.ts

```diff
@@ -8132,10 +8132,12 @@ namespace ts {
         directoryExists?(path: string): boolean;
         readFile?(path: string): string | undefined;
         realpath?(path: string): string;
-        getSymlinkCache?(): SymlinkCache;
+        getSymlinkCache?(): SymlinkCache | undefined;
+        getResolvedTypeReferenceDirectives?(): ReadonlyESMap<string, ResolvedTypeReferenceDirective | undefined>;
         getModuleSpecifierCache?(): ModuleSpecifierCache;
         getGlobalTypingsCacheLocation?(): string | undefined;
         getNearestAncestorDirectoryWithPackageJson?(fileName: string, rootDir?: string): string | undefined;
+        preferFailureToResultsWithIgnoredPaths?(): boolean;

```

Comment:
why does this need to be function instead of just optional value  and what this is achieving..
Can you comment here what this is to mean?

Reply:
I’m going to revert this change from this PR for now to unblock it, since I’m not happy with the name and haven’t come up with anything better. It isn’t strictly related to the primary goal of this PR; it just finishes the job of filtering out pnpm paths that have no non-pnpm path symlinks. This is a lower priority fix since it doesn’t block any good auto imports.